crates/router/src/core/payments/operations/payment_session.rs

```diff
@@ -310,131 +308,93 @@ where
-            .filter(|connector_account| {
-                connector_account
-                    .payment_methods_enabled
-                    .clone()
-                    .unwrap_or_default()
-                    .iter()
-                    .any(|payment_method| {
-                        let parsed_payment_method_result: Result<
-                            PaymentMethodsEnabled,
-                            error_stack::Report<errors::ParsingError>,
-                        > = payment_method.clone().parse_value("payment_method");
-
-                        match parsed_payment_method_result {
-                            Ok(parsed_payment_method) => parsed_payment_method
-                                .payment_method_types
-                                .map(|payment_method_types| {
-                                    payment_method_types.iter().any(|payment_method_type| {
-                                        matches!(
-                                            payment_method_type.payment_method_type,
-                                            api_models::enums::PaymentMethodType::GooglePay
-                                        )
-                                    })
-                                })
-                                .unwrap_or(false),
-                            Err(parsing_error) => {
-                                logger::debug!(session_token_parsing_error=?parsing_error);
-                                false
+        let mut connector_and_supporting_wallet = Vec::new();
+
+        for connector_account in connector_accounts {
+            let payment_methods = connector_account
+                .payment_methods_enabled
+                .unwrap_or_default();
+            for payment_method in payment_methods {
+                let parsed_payment_method_result: Result<
+                    PaymentMethodsEnabled,
+                    error_stack::Report<errors::ParsingError>,
+                > = payment_method.clone().parse_value("payment_method");
+
+                match parsed_payment_method_result {
+                    Ok(parsed_payment_method) => {
+                        let payment_method_types = parsed_payment_method
+                            .payment_method_types
+                            .unwrap_or_default();
+                        for payment_method_type in payment_method_types {
+                            if matches!(
+                                payment_method_type.payment_experience,
+                                Some(api_models::enums::PaymentExperience::InvokeSdkClient)
+                            ) {
+                                let connector_and_wallet = (
+                                    connector_account.connector_name.to_owned(),
+                                    payment_method_type.payment_method_type.to_string(),
+                                );
+                                connector_and_supporting_wallet.push(connector_and_wallet);
```

Comment:
This can be done using iterators. 
```suggestion
let connector_and_supporting_wallet = connector_accounts
            .iter()
            .map(|connector_account| {
                connector_account
                    .payment_methods_enabled
                    .unwrap_or_else(Vec::new)
                    .iter()
                    .map(|payment_methods_enabled| {
                        let parsed_payment_method_result: Result<
                            PaymentMethodsEnabled,
                            _,
                        > = payment_methods_enabled
                            .clone()
                            .parse_value("payment_method");

                        parsed_payment_method_result
                            .ok()
                            .map(|payment_methods_enabled| {
                                payment_methods_enabled
                                    .payment_method_types
                                    .unwrap_or_default()
                                    .iter()
                                    .filter(|payment_method| {
                                        matches!(
                                            payment_method.payment_experience,
                                            Some(api_models::enums::PaymentExperience::InvokeSdkClient)
                                        )
                                    }).map(|filtered_payment_method| {
                                        (connector_account.connector_name, api::GetToken::Connector)
                                    })
                            }).unwrap_or(std::iter::empty()).collect::<Vec<_>>()
                    })
            })
            .flatten().flatten().collect::<Vec<_>>();
```